src/fu-engine.c

```diff
@@ -3758,6 +3759,71 @@ fu_engine_get_item_by_id_fallback_history(FuEngine *self, const gchar *id, GErro
 	return NULL;
 }
 
+static gboolean
+fu_engine_search_query_append(FuEngine *self, const gchar *xpath, GError **error)
+{
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(XbQuery) query = NULL;
+
+	/* prepare tag query with bound GUID parameter */
+	query = xb_query_new_full(self->silo, xpath, XB_QUERY_FLAG_OPTIMIZE, &error_local);
+	if (query == NULL) {
+		if (g_error_matches(error_local, G_IO_ERROR, G_IO_ERROR_NOT_FOUND) ||
+		    g_error_matches(error_local, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT)) {
+			g_debug("ignoring prepared query %s: %s", xpath, error_local->message);
+			return TRUE;
+		}
+		g_propagate_error(error, g_steal_pointer(&error_local));
+		fwupd_error_convert(error);
+		return FALSE;
+	}
+	g_ptr_array_add(self->search_queries, g_steal_pointer(&query));
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_engine_search_query_create(FuEngine *self, GError **error)
+{
+	/* invalidate everything */
+	g_ptr_array_set_size(self->search_queries, 0);
+
+	/* we get one for free, add build the others */
+	g_ptr_array_add(self->search_queries, g_object_ref(self->query_component_by_guid));
+	if (!fu_engine_search_query_append(self, "components/component/id[text()=?]/..", error))
+		return FALSE;
+	if (!fu_engine_search_query_append(self, "components/component/name[text()~=?]/..", error))
+		return FALSE;
+	if (!fu_engine_search_query_append(self,
+					   "components/component/developer_name[text()~=?]/..",
+					   error))
+		return FALSE;
+	if (!fu_engine_search_query_append(self,
+					   "components/component/releases/release/artifacts/"
+					   "artifact/filename[text()=?]/../../../../..",
+					   error))
+		return FALSE;
+	if (!fu_engine_search_query_append(self,
+					   "components/component/releases/release/artifacts/"
+					   "artifact/checksum[text()=?]/../../../../..",
+					   error))
+		return FALSE;
+	if (!fu_engine_search_query_append(
+		self,
+		"components/component/releases/release/issues/issue[text()=?]/../../../..",
+		error))
+		return FALSE;
```

Comment:
Do these work? I can't get it to return anything.

Reply:
I added some examples here: https://github.com/fwupd/fwupd/commit/b2d84915698747e73de0b654404612bcd4d1b400#diff-6f384dfb346d0114831241701b38e58a95ea084fad39b5501a62ba85146c2a82R283-R284

Reply:
I mean specifically these three that I highlighted in the comment. I think your examples don't include those.